pkg/orchestrator/housekeeping_test.go

```diff
@@ -0,0 +1,343 @@
+//go:build unit || !integration
+
+package orchestrator
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/bacalhau-project/bacalhau/pkg/jobstore"
+	"github.com/bacalhau-project/bacalhau/pkg/models"
+	"github.com/bacalhau-project/bacalhau/pkg/test/mock"
+	"github.com/stretchr/testify/suite"
+	"go.uber.org/mock/gomock"
+)
+
+const (
+	mockJobTimeout                 = 60              // 1 minute
+	timeoutBuffer                  = 1 * time.Minute // another minute. only executions older than 2 minutes are expired
+	notExpiredModifyTime           = -30 * time.Second
+	expiredWithinBufferModifyTime  = -90 * time.Second
+	expiredOutsideBufferModifyTime = -120 * time.Second
+)
+
+type HousekeepingTestSuite struct {
```

Comment:
Strongly prefer we test things involving time with github.com/benbjohnson/clock as we have done elsewhere in the code base. This way we can avoid depending on time in tests, preventing flakes, and also improve how tests behave when using a debugger: Currently these tests will fail if a debugger is used to inspect them as the timeouts are out of our control.

Reply:
I am a big fan of controlling the clock for testing purposes. Though in this case it will require that we pass the clock to `Execution.IsExpired(...)` which I am not a fan of. Still, the biggest advantage of mocking the clock is to have more predictable test scenarios and to avoid having to sleep here and there to mimic some scenarios. This is not the case here as I am explicitly setting the executions time. I am only using the wall clock to check if those times have expired or not. 

I agree that if you put a break point in very specific places the tests will fail, but I don't see this as a problem. 

In summary, I like clocks, but for this specific test suite I don't see they add significant value and justify  passing the clock to `Execution.IsExpired(...)`

Reply:
> Though in this case it will require that we pass the clock to Execution.IsExpired(...) which I am not a fan of

Or we could pass the time at which the execution ought to expire instead of a duration as we do now. Actually I might like that better as it removes time from the model. Wdyt?